realm/realm-library/src/androidTest/java/io/realm/RealmIntegerTests.java

```diff
@@ -0,0 +1,326 @@
+
+
+@RunWith(AndroidJUnit4.class)
+public class RealmIntegerTests {
+
+    @Rule
+    public final TestRealmConfigurationFactory configFactory = new TestRealmConfigurationFactory();
+    @Rule
+    public final RunInLooperThread looperThread = new RunInLooperThread();
+
+    private Realm realm;
+
+    @Before
+    public void setUp() throws Exception {
+        RealmConfiguration realmConfig = configFactory.createConfiguration();
+        realm = Realm.getInstance(realmConfig);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        if (realm != null) {
+            if (realm.isInTransaction()) {
+                realm.cancelTransaction();
+            }
+            realm.close();
+        }
+    }
+
+    /**
+     * Validate basic constructor parameters and functions set, increment and decrement.
+     */
+    @Test
+    public void basic_basic() {
+        RealmInteger ri1 = new RealmInteger(10);
+        RealmInteger ri2 = new RealmInteger("10");
+        assertEquals(ri1, ri2);
+
+        ri1.set(15);
+        ri1.decrement(2);
+        ri2.increment(3);
+        assertEquals(ri1, ri2);
+    }
+
+    /**
+     * Validate various getters.
+     * Expected behaviour is that gets of smaller sized quantities should truncate on the left:
+     * that they should return exactly the rightmost N bits of the underlying value.
+     * <p>
+     * Caution.  The assertion functions will cast back up to int
+     * if either arg is an int.  That will cause sign extension.
+     */
+    @Test
+    public void basic_getters() {
+        RealmInteger ri = new RealmInteger(0x5555444433332211L);
+
+        // positive
+        assertEquals(0x5555444433332211L, ri.longValue());
+        assertEquals(0x033332211, ri.intValue());
+        assertEquals((short) 0x02211, ri.shortValue());
+        assertEquals((byte) 0x011, ri.byteValue());
+
+        assertEquals(6.1488962E18F, ri.floatValue());
+        assertEquals(6.1488959259517348E18, ri.doubleValue());
+
+        // negative
+        ri.set(0x8888444483338281L);
+        assertEquals(0x8888444483338281L, ri.longValue());
+        assertEquals(0x083338281, ri.intValue());
+        assertEquals((short) 0xf8281, ri.shortValue());
+        assertEquals((byte) 0xf81, ri.byteValue());
+
+        assertEquals(-8.6085554E18F, ri.floatValue());
+        assertEquals(-8.6085556266690468E18, ri.doubleValue());
+    }
+
+    /**
+     * Be absolutely certain that we can actually compare two longs.
+     */
+    @Test
+    public void basic_compareTo() {
+        RealmInteger ri1 = new RealmInteger(10);
+        RealmInteger ri2 = new RealmInteger("11");
+
+        assertEquals(ri1.compareTo(ri2), -1);
+        ri2.decrement(1);
+        assertEquals(ri1.compareTo(ri2), 0);
+        ri2.decrement(1);
+        assertEquals(ri1.compareTo(ri2), 1);
+    }
+
+    /**
+     * Be absolutely certain that this is a long.
+     */
+    @Test
+    public void basic_underflowAndOverflow() {
```

Comment:
I would split this in two: "increment_underFlowAndOverflow()" and `decrement_underFlowAndOverflow()", which fits the pattern we use elsewhere better.